svmbir/utils.py

```diff
@@ -0,0 +1,42 @@
+import numpy as np
+import matplotlib.pyplot as plt
+def gen_ellipse(x0,y0,a,b,value,pre_img,theta=0):
+    """
+    Generate ellipse in 2D plane.
+    
+    Args: 
+        x0(float): X-axis coordinate.
+        y0(float): Y-axis coordinate.
+        a(float): X-axis radius.
+        b(float): Y-axis radius.
+        value(float): Value of the ellipse.
+        pre_img(float): Plane that the ellipse added to. 
+        theta(float): Angle of rotation, from -PI to PI.
+
+
+    Return:
+        ndarray: 2D array with the same shape of pre_img.
+
+    """    
+    pre_img[((x-x0)*np.cos(theta)+(y-y0)*np.sin(theta))**2/a**2 \
+    +((x-x0)*np.sin(theta)-(y-y0)*np.cos(theta))**2/b**2<=1]=value
+
```

Comment:
x and y are defined in the script (line 33) rather than in this function.  
*** So this function is not standalone and will not do what you expect it to do in general. *** 
Please write a specification about what input and output you want to provide.  Are you going to define the ellipse in terms of the indices of the array or in terms of units like you've defined in x and y in line 33 or standard units like [-1,1] for each axis?  Please look at https://github.com/mckib2/phantominator/blob/master/phantominator/ct_shepp_logan.py
for an example of how to do the documentation and specification.  You don't have to do that exactly, but it should be that clear.  Please provide a more detailed description of the function above Args: and make a distinction between  scalars and arrays in your args.

Reply:
Why are you importing matplotlib into this module? Is this required?